pkg/utils/utils.go

```diff
@@ -383,15 +375,15 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 	return nil
 }
 
-func generateRandomMacAddressVfs() (string, error) {
+func generateRandomMacAddressVfs() ([]byte, error) {
 	buf := make([]byte, 6)
 	_, err := rand.Read(buf)
```

Comment:
Some MAC addresses are restricted. E.g. IEEE802.1 reserved MACs.
If we are unlucky and somehow land on restricted MAC addresses, we shouldn't use those MAC address for the interface.

Reply:
OR even multicast MAC addresses shouldn't be allowed.

Reply:
Yes, we can go very far with this and also disallow MACs that have been generated for other VFs. I think if we just fix the first byte to some hardcoded value, then we are ok.

Reply:
[We or the first byte with ](https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/325/files#diff-81ddbadfb415ccbb9c7af84f11668d1aa5e53c34025bf86d4702f16b4e42f045R385)2 and guarantee the local bit is set so we know we have a unicast address.

Is there anything else we need to do here?

Reply:
looks ok, thanks